src/ros_model_generator/rossystem_generator.py

```diff
@@ -16,11 +16,146 @@
+import ros_metamodels.ros_metamodel_core as model
+import rosgraph
+import rosparam
+import rosservice
+
+
+BLACK_LIST_PARAM = ['/rosdistro', '/rosversion', '/run_id']
+BLACK_LIST_TOPIC = ["/tf", "/tf_static", "/rosout", "/clock"]
+BLACK_LIST_SERV = ["/set_logger_level", "/get_loggers"]
+BLACK_LIST_NODE = ["/rosout"]
+
+ACTION_FILTER = ['goal', 'cancel']
+ACTION_FILTER2 = ['status', 'result', 'feedback']
+
+def check_black_list(name, black_list):
+    for bl_ in black_list:
+        if bl_ in name:
+            return False
+    return True
+
+def init_node_dict(nodes, name):
+    nodes[name] = {'publishers' : dict(),
+                   'subscribers' : dict(),
+                   'service_servers' : dict(),
+                   'service_clients' :dict(),
+                   'action_servers' :dict(),
+                   'action_clients' : dict() }
+
+def check_actions(publishers, subscribers, action_clients, action_servers):
+        pubs_ = [pub for pub in publishers.keys()]
+        subs_ = [sub for sub in subscribers.keys()]
+
+        remove_pubs = list()
+        remove_subs = list()
+
+        # Check Action client
+        for topic_name, topic_type in publishers.items():
+            if topic_name.endswith(ACTION_FILTER[0]):
+                _action_name = topic_name[:-len(ACTION_FILTER[0]) - 1]
+                cancel_topic = _action_name + '/' + ACTION_FILTER[1]
+                if not (cancel_topic in pubs_):
+                    continue
+                remove_pubs.append(topic_name)
+                remove_pubs.append(cancel_topic)
+                for name in ACTION_FILTER2:
+                    topic = _action_name + '/' + name
+                    if not (topic in subs_):
+                        continue
+                    remove_subs.append(topic)
+                _action_type = topic_type[:-10]  # Hardcoded ActionGoal
+                action_clients.add((_action_name, _action_type))
+
+        # Check Action Server
+        for topic_name, topic_type in subscribers.items():
+            if topic_name.endswith(ACTION_FILTER[0]):
+                _action_name = topic_name[:-len(ACTION_FILTER[0]) - 1]
+                cancel_topic = _action_name + '/' + ACTION_FILTER[1]
+                if not (cancel_topic in subs_):
+                    continue
+                remove_subs.append(topic_name)
+                remove_subs.append(cancel_topic)
+                for name in ACTION_FILTER2:
+                    topic = _action_name + '/' + name
+                    if not (topic in pubs_):
+                        continue
+                    remove_pubs.append(topic)
+                _action_type = topic_type[:-10]  # Hardcode ActionGoal
+                action_servers[_action_name] = _action_type
+
+        for topic in remove_pubs:
+            publishers.pop(topic)
+        for topic in remove_subs:
+            subscribers.pop(topic)
+
+def create_ros_graph_snapshot():
```

Comment:
Ops, I think I misunderstood something. This code shouldn't be here. 

The `ros_model_parser`,  specially the parser and generator classes, should be independent of the ROS version as it will be installed as a python module for the extractors. Ideally we don't have dependencies to python ROS API packages. Fits this function somewhere within the rosgraph_monitor repository?

Reply:
Sure. I can revert this and https://github.com/ipa320/rosgraph_monitor/pull/1/commits/9873aead7b2249ffe62404284878af6a7ba49412 commits.

Reply:
@ipa-nhg reverted. Will work on params today and then should be good to be merged.